### PR TITLE
Feature - Input: Add <C-w> and <C-h> handling to the input model

### PR DESCRIPTION
**Pull request by bryphe**
This adds support for `<C-h>` (which is like backspace) and `<C-w>` (delete word) gestures to the `InputModel`. I use these often in insert mode and I miss them in the search and quickopen experiernce.

### CODE REVIEW

File: test/Components/InputModelTests.re [hunk 1]
```diff
@@ -228,23 +228,21 @@ describe("handleInputS#handleInput", ({describe, _}) => {
...
 
       expect.equal(expected, newSelection);
       expect.string(text).toEqual(
-        "Som interesting. Test. String. Isn't it? Maybe",
+        " interesting. Test. String. Isn't it? Maybe",
       );
     });
 
-    test("Doesn't remove character if cursor at the beginnig", ({expect, _}) => {
+    test("Doesn't remove anything if cursor at the beginnig", ({expect, _}) => {
```
Comment (glennsl) [inline test/Components/InputModelTests.re]:
```suggestion
    test("Doesn't remove anything if cursor at the beginning", ({expect, _}) => {
```

File: test/Components/InputModelTests.re [hunk 2]
```diff
@@ -265,6 +263,48 @@ describe("handleInputS#handleInput", ({describe, _}) => {
...
+           let (text, newSelection) = runInputHandler(selection, key);
+
+           expect.equal(expected, newSelection);
+           expect.string(text).toEqual(
+             "Som interesting. Test. String. Isn't it? Maybe",
+           );
+         });
+
+         test(
+           "Doesn't remove character if cursor at the beginnig",
```
Comment (glennsl) [inline test/Components/InputModelTests.re]:
```suggestion
           "Doesn't remove character if cursor at the beginning",
```

(I can apparently see typos now, just as long as they're not mine)

File: test/Components/InputModelTests.re [hunk 3]
```diff
@@ -265,6 +263,48 @@ describe("handleInputS#handleInput", ({describe, _}) => {
...
+    |> List.iter(key => {
+         test("Removes character on the left of cursor", ({expect, _}) => {
+           let selection = collapsedSelection(4);
+           let expected = collapsedSelection(3);
+
+           let (text, newSelection) = runInputHandler(selection, key);
+
+           expect.equal(expected, newSelection);
+           expect.string(text).toEqual(
+             "Som interesting. Test. String. Isn't it? Maybe",
```
Comment (glennsl) [inline test/Components/InputModelTests.re]:
```suggestion
             "Some interesting. Test. String. Isn't it? Maybe",
```

File: test/Components/InputModelTests.re (hunk 1 above)
Comment (bryphe) [reply test/Components/InputModelTests.re]: Nice catch! Thanks @glennsl

File: test/Components/InputModelTests.re (hunk 2 above)
Comment (bryphe) [reply test/Components/InputModelTests.re]: 😆 

We need to bring in a spell-checker extension! Would be a good test case for the exthost

**Pull request merged**